app/api/src/docker/cadquery/cadquery.js

```diff
@@ -1,53 +1,55 @@
 const { runCQ } = require('./runCQ')
 const middy = require('middy')
 const { cors } = require('middy/middlewares')
+const AWS = require('aws-sdk')
+const {
+  makeHash,
+  checkIfAlreadyExists,
+  getObjectUrl,
+  loggerWrap,
+  storeAssetAndReturnUrl,
+} = require('../common/utils')
 
-// cors true does not seem to work in serverless.yml, perhaps docker lambdas aren't covered by that config
-// special lambda just for responding to options requests
-const preflightOptions = (req, _context, callback) => {
-  const response = {
-    statusCode: 204,
-    headers: {
-      'Access-Control-Allow-Origin': '*',
-      'Access-Control-Allow-Methods': 'POST',
-      'Access-Control-Allow-Headers': '*',
-    },
-  }
-  callback(null, response)
-}
+const s3 = new AWS.S3()
 
 const stl = async (req, _context, callback) => {
   _context.callbackWaitsForEmptyEventLoop = false
-  const eventBody = Buffer.from(req.body, 'base64').toString('ascii')
-  console.log(eventBody, 'eventBody')
-  const { file, settings } = JSON.parse(eventBody)
-  const { error, result, tempFile } = await runCQ({ file, settings })
-  if (error) {
-    const response = {
-      statusCode: 400,
-      body: JSON.stringify({ error, tempFile }),
-    }
-    callback(null, response)
-  } else {
-    console.log(`got result in route: ${result}, file is: ${tempFile}`)
-    const fs = require('fs')
-    const image = fs.readFileSync(`/tmp/${tempFile}/output.stl`, {
-      encoding: 'base64',
-    })
-    console.log(image, 'encoded image')
+  const eventBody = req.body
+  console.log('eventBody', eventBody)
+  const key = `${makeHash(eventBody)}.stl`
+  console.log('key', key)
+
+  const params = {
+    Bucket: process.env.BUCKET,
+    Key: key,
+  }
+  const previousAsset = await checkIfAlreadyExists(params, s3)
+  if (previousAsset.isAlreadyInBucket) {
```

Comment:
Return early if the asset is already in the bucket. (based on the key which is a hash of the user's code)